documentation/user_docs/docs/manual/Advanced_Multifit.rst

```diff
@@ -2,133 +2,160 @@
 
 
 Utility methods
 ---------------
 
-These are required to enable multifit to work with objects
+These are methods ``multifit`` can use to fit user-defined objects
+
+mask
+~~~~
+
+.. code-block:: matlab
+
+   wout = mask(win, msk)
+
+
+A method that masks data points from further calculation. The output object must
+be a valid instance of the class in which the masked values have been removed in
+whatever sense the class requires.
+
+.. _mask_points:
 
-wout = mask(win, msk)
+mask_points [optional]
 ~~~~~~~~~~~~~~~~~~~~~~
 
-A method that masks data points from further calculation. The output object must be a valid instance of the class in
-which the masked values have been removed in whatever sense the class requires.
+.. code-block:: matlab
 
+   [msk, ok, mess] = mask_points(win, 'keep', xkeep, 'remove', xremove, 'mask', msk_in)
 
-[y,var,msk] = sigvar_get(win)
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Create a mask array given ranges of x-coordinates to keep, remove or mask from the array. The elements of a mask array are
+``true`` for those data points which are to be retained.
+
+Function must output a logical flag ``ok``, with message string if ``ok==false`` rather than terminate.
 
-A method that returns the intensity and variance arrays from the objects, along with a mask array that indicates which
-elements are to be retained (where elements of msk are true, the corresponding elements of y and var are retained). The
-output arrays y and var must have the same size and shape; msk must have the same number of elements (but can be a
-different shape). The array msk must be understood by the method 'mask' defined below.
+(It is possible to have the function terminate if ``ok`` and ``mess`` are not given as return arguments; it is the
+advanced syntax that is required within ``multifit``).
+
+sigvar_get
+~~~~~~~~~~
 
+.. code-block:: matlab
 
-wsum = w1 + w2
-~~~~~~~~~~~~~~
+   [y, var, msk] = sigvar_get(win)
 
-If a background function is provided, addition of objects must be defined (requires overloading of the addition
-operator with a method named plus.m).
+A method that returns the intensity and variance arrays from the objects, along
+with a mask array that indicates which elements are to be retained (where
+elements of msk are true, the corresponding elements of y and var are
+retained). The output arrays y and var must have the same size and shape; msk
+must have the same number of elements (but can be a different shape). The array
+msk must be understood by the method 'mask' defined below.
 
+.. _sigvar_getx:
+
+sigvar_getx [optional]
+~~~~~~~~~~~~~~~~~~~~~~
 
 
-x = sigvar_getx(win) [optional]
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+.. code-block:: matlab
 
-Get the corresponding x values to the y, var, msk arrays that are returned by sigvar_get.
+   x = sigvar_getx(win)
+
+Get the corresponding ``x`` values to the ``y``, ``var``, ``msk`` arrays that
+are returned by ``sigvar_get``.
```

Comment:
what does optional mean? How does sigvar_getx know about y,var and msk? What is win?